test/e2e/autoscale_test.go

```diff
@@ -257,7 +270,20 @@ func assertScaleDown(ctx *testContext) {
 	ctx.t.Log("Scaled down.")
 }
 
-func numberOfPods(ctx *testContext) (float64, error) {
+func allPods(ctx *testContext) ([]corev1.Pod, error) {
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{serving.RevisionLabelKey: ctx.resources.Revision.Name}}
```

Comment:
This selector definition seems redundant. Seems like you can just move the `map[string]string` into the `labels.Set` call below directly.